presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoPreparedStatement.java

```diff
@@ -479,7 +495,9 @@ public void setArray(int parameterIndex, Array x)
     public ResultSetMetaData getMetaData()
             throws SQLException
     {
-        throw new SQLFeatureNotSupportedException("getMetaData");
+        DatabaseMetaData databaseMetaData = super.getConnection().getMetaData();
+        ResultSet resultSet = ((PrestoDatabaseMetaData) databaseMetaData).getDescribeOutput(statementName);
+        return new PrestoResultSetMetaData(getDescribeOutputColumnInfoList(resultSet));
```

Comment:
1. use try-with-resources to ensure `ResultSet` (and temporary `Statement` if needed) are closed -- the returned `PrestoResultSetMetaData` does not hold on to the `ResultSet` being obtained here

2. It's unclear from the javadoc whether `getMetaData()` can be called after `execute` was called.

If it cannot, then you can simply do this here:
```
try (ResultSet resultSet = super.executeQuery("DESCRIBE OUTPUT " + statementName)) {
    return new PrestoResultSetMetaData(getDescribeOutputColumnInfoList(resultSet));
}
```

If it can, we need a separate, temporary statement object:
```
try (Statement statement = connection().createStatement();
        ResultSet resultSet = statement.executeQuery("DESCRIBE OUTPUT " + statementName)) {
    return new PrestoResultSetMetaData(getDescribeOutputColumnInfoList(resultSet));
}
```

@electrum do you know?

Reply:
I'm not sure either, but it should be fine either way since this is our code.

Reply:
Given https://github.com/prestosql/presto/blob/master/presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoPreparedStatement.java#L657, the first one won't be able to run. I will just use the second approach.